### DISCUSSION TITLE
Ignoring System.Threading.Tasks.TaskCanceledException

### DISCUSSION THREAD

**Discussion by paurail** [imperugo/SQS.Extensions#137] [Ideas]
During a redeploy  in prod we found lots of fatal errors that trigger alerts. I think we should ignore  TaskCanceledException in base SQS class (_SqsHostedService_). or at least allow to override how errors are handled to prevent such problems? Don't mind creating a PR.

**Comment by imperugo** [comment]
Hi @paurail could you provide me more information about the issue your are getting?

**Comment by paurail** [comment]
@imperugo its about Errors that are logged by SqsHostedService. It catches all exceptions and logs as Errors or Critical. Even when they are not significant ones: e.g TaskCancelledException

**Reply by imperugo** [reply]
ok, understood your point. 

I'll do something

**Comment by imperugo** [comment]
I've just pushed and update that should be handle the issue.

Here https://github.com/imperugo/SQS.Extensions/blob/main/src/SQS.Extensions/HostedServices/SqsHostedService.cs#L67-L74 the change
